common/src/main/java/com/wynntils/core/functions/FunctionManager.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.functions;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.managers.CoreManager;
+import com.wynntils.functions.TestFunction;
+import com.wynntils.functions.WorldNameFunction;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TextComponent;
+
+/** Manage all built-in {@link Function}s */
+public final class FunctionManager extends CoreManager {
+    private static final List<Function> FUNCTIONS = new ArrayList<>();
+    private static final Set<EnableableFunction> ENABLED_FUNCTIONS = new HashSet<>();
+
+    private static void registerFunction(Function function) {
+        FUNCTIONS.add(function);
+        if (function instanceof EnableableFunction<?> enableableFunction) {
+            enableableFunction.init();
+        }
+    }
+
+    public static List<Function> getFunctions() {
+        return FUNCTIONS;
+    }
+
+    public static boolean enableFunction(Function function) {
+        if (!(function instanceof EnableableFunction<?> enableableFunction)) return true;
```

Comment:
Why not take EnableableFeature as arg? Also I can't see where we call this

Reply:
Good point about the argument. I'll fix.

This is currently called when doing `/wynntils function enable <function>`. My thinking is that it should also be called when an EnableableFunction is added to an info box overlay.

Reply:
Actually, about the argument, if I change that, that will force the caller to know about what kind of function it is. If I keep it, the info box can always call "enable" and know they are doing the right thing. So I think I'll prefer to keep it this way, if that is OK with you?

Reply:
okay either way